pkg/machinery/resources/network/link_spec.go

```diff
@@ -66,8 +66,9 @@ var (
 //
 //nolint:gocyclo
 func (spec *LinkSpecSpec) Merge(other *LinkSpecSpec) error {
-	if spec.Logical != other.Logical {
-		return fmt.Errorf("mismatch on logical for %q: %v != %v", spec.Name, spec.Logical, other.Logical)
+	// prefer Logical, as it is defined for bonds/vlans, etc.
```

Comment:
is it possible to have tests for this?

Reply:
yes, I can add tests, even though the code here is relatively trivial, we probably can do some higher level in the actual controller which merges the specs

Reply:
added the test, thanks for the review!